### ISSUE TITLE
AsyncDruidWebServiceImpl tries to stop nonexistent timer for DruidDimensionsLoader

### ISSUE THREAD

**Issue by kevinhinterlong** [yahoo/fili#314]
In `DruidDimensionsLoader` it posts a query to druid but no timer is started for the `RequestLog`.
```java
druidWebService.postDruidQuery(requestContext, success, errorCallback, failureCallback, druidSearchQuery); // line 176
```

In `postDruidQuery(...)` it then tries to stop a nonexistent timer. 

```java
if (!(druidQuery instanceof WeightEvaluationQuery)) { // line 371
    if (context.getNumberOfOutgoing().decrementAndGet() == 0) {
        RequestLog.stopTiming(REQUEST_WORKFLOW_TIMER);
    }
    ...
```

Should there be a call to start the timer in `DruidDimensionsLoader` or check if there’s an active timer in `AsyncDruidWebServiceImpl`?

**Comment by cdeszaq** [comment]
Yeah, this is a problem we've tried hard to ignore. The `AsyncDruidWebServiceImpl` is, as you've found, assuming it's being called as part of a request workflow. I'm not sure how best to make it more general, but one idea would be to pass in the timers that the caller would want it to stop/start after it makes the call / picks up the response. 

Other options would be to have more instances of the client, or to split the client into more layers so that there can be a layer for the logic around start/stop of queries, and a layer for just the mechanics of request/response, but those have other impacts as well. More ideas for a solution would be good 😄

**Comment by kevinhinterlong** [comment]
I think passing in the timers would be a good idea, but with the addition of a sql backend (and others in the future), I think it might make sense to split the client and have one of them do the timings.

Then we could make an interface or some contract that the backends must fulfill for starting/stopping certain timers. 

Related: #336
